lua/mapvote/sv_mapvote.lua

```diff
@@ -87,127 +87,74 @@ local function mapVoteOver( autoGamemode, callback )
 
     local map = MapVote.CurrentMaps[winner]
 
-    local gamemode = nil
-
-    if autoGamemode then
-        -- check if map matches a gamemode's map pattern
-        for _, gm in pairs( engine.GetGamemodes() ) do
-            -- ignore empty patterns
-            if gm.maps and gm.maps ~= "" then
-                -- patterns are separated by "|"
-                for _, pattern in pairs( string.Split( gm.maps, "|" ) ) do
-                    if string.match( map, pattern ) then
-                        gamemode = gm.name
-                        break
-                    end
-                end
-            end
-        end
-    else
-        print( "not enabled" )
-    end
-
     timer.Simple( 4, function()
-        if ( hook.Run( "MapVoteChange", map ) ~= false ) then
-            if ( callback ) then
-                callback( map )
-            else
-                -- if map requires another gamemode then switch to it
-                if ( gamemode and gamemode ~= engine.ActiveGamemode() ) then
-                    RunConsoleCommand( "gamemode", gamemode )
-                end
-                RunConsoleCommand( "changelevel", map )
-            end
+        if hook.Run("MapVoteChange", map) == false then return end
+        if callback then
+            callback( map )
         end
+
+        RunConsoleCommand( "changelevel", map )
```

Comment:
In the original code it runs either callback or the console command, in this pr it does both. Not sure if thats intentional

Reply:
it is

Reply:
this behavior makes more sense and we dont use the callback for anything anyway